core/src/main/java/org/frankframework/util/RestoreMovedElementsHandler.java

```diff
@@ -15,61 +15,108 @@
 */
 package org.frankframework.util;
 
-import lombok.NoArgsConstructor;
+import lombok.Setter;
 import lombok.extern.log4j.Log4j2;
 import org.frankframework.core.PipeLineSession;
+import org.frankframework.xml.FullXmlFilter;
+import org.xml.sax.Attributes;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
 
+/**
+ * Restore moved elements from pipelineSession.
+ * Requires a session with the moved elements stored as key-value pairs.
+ * The output of the XMLWriter is data with moved elements restored, if they were found.
+ */
 @Log4j2
-@NoArgsConstructor(access = lombok.AccessLevel.PRIVATE)
-public class RestoreMovedElementsHandler {
-
+public class RestoreMovedElementsHandler extends FullXmlFilter {
 	private static final String ME_START = "{sessionKey:";
 	private static final String ME_END = "}";
+	private boolean inCDATASection = false;
+
+	private final StringBuilder charDataBuilder = new StringBuilder();
+	private @Setter PipeLineSession session;
+
+	public RestoreMovedElementsHandler(ContentHandler handler) {
+		super(handler);
+	}
+
+	@Override
+	public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+		processCharacterData(true);
+		super.startElement(uri, localName, qName, attributes);
+	}
+
+	@Override
+	public void endElement(String uri, String localName, String qName) throws SAXException {
+		processCharacterData(false);
+		super.endElement(uri, localName, qName);
+	}
+
+	@Override
+	public void characters(char[] ch, int start, int length) throws SAXException {
+		charDataBuilder.append(ch, start, length);
```

Comment:
Poeh, ik had hier eingelijk de find/replace verwacht. Niet in de start- en endElement methodes.

Reply:
Ja, dat had ik ook, toen ik begon. En dan ga je het bouwen en zie je dat dat niet werkt. Je moet juist bij het starten van een element alle charData verzamelen en controleren van wat er voor ontvangen is.